### PR TITLE
fix(log): set info log level by default for the entire application

### PR DESCRIPTION
**Pull request by bnjjj**
By default the log level of the application if `RUST_LOG` is not set is `info` and if you provide `--log` or `APOLLO_RUST_LOG` value then it overrides the default `info` log level with `apollo_router=...` because it only impacts the `apollo_router` crate, not external custom plugin and so one.

By doing this fix, by default if you have a custom plugin with info logs or metrics you won't have to enforce `RUST_LOG=info` everytime, it will work as expected.

> Note: it doesn't impact the behavior of `RUST_LOG` if you set it.

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [x] Changes are compatible[^1]
- [x] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [x] Unit Tests
    - [ ] Integration Tests
    - [x] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

**Review by o0Ignition0o — APPROVED** [review]

File: .changesets/fix_bnjjj_fix_log_level.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+### Set log level to `info` by default for the entire application ([PR #4451](https://github.com/apollographql/router/pull/4451))
+
+By default the log level of the application if `RUST_LOG` is not set is `info` and if you provide `--log` or `APOLLO_RUST_LOG` value then it overrides the default `info` log level with `apollo_router=...` because it only impacts the `apollo_router` crate, not external custom plugin and so one.
+
+By doing this fix, by default if you have a custom plugin with info logs or metrics you won't have to enforce `RUST_LOG=info` everytime, it will work as expected.
```
Comment (o0Ignition0o) [inline .changesets/fix_bnjjj_fix_log_level.md:5]:
```suggestion
Customizing the router's [Log level](https://www.apollographql.com/docs/router/configuration/telemetry/exporters/logging/overview/#log-level) now applies to the router *and* native plugins.
```

Comment (bnjjj) [reply .changesets/fix_bnjjj_fix_log_level.md:5]: That's wrong, it doesn't apply to the plugins if you set `debug`. This change introduces a better default behavior. To not have to enforce `RUST_LOG=info`

**Review by BrynCooke — CHANGES_REQUESTED** [review]
The docs need to updated:
https://www.apollographql.com/docs/router/configuration/telemetry/exporters/logging/overview/#log-level

Also @garypen do you agree with the overall change?

**Comment by bnjjj** [comment]
@BrynCooke I think the documentation is still accurate, I didn't change the behavior. I just fixed it

**Comment by garypen** [comment]
> The docs need to updated: https://www.apollographql.com/docs/router/configuration/telemetry/exporters/logging/overview/#log-level
> 
> Also @garypen do you agree with the overall change?

The motivation of the last adjustment to APOLLO_ROUTER_LOG was to limit its scope to "just router stuff". We restored RUST_LOG functionality to what a rust developer would expect (i.e.: you can do whatever is supported by the RUST_LOG filtering). I think there was an oversight in that fix which mean that APOLLO_ROUTER_LOG didn't apply to plugins, so if you wanted to see logs from plugins you had to use RUST_LOG.

If this fix is addressing that shortcoming, i.e.: plugins observe the same logging behaviour as the main router, then that is correct.

It looks to me like it might also change the value for non-plugin crates, e.g.: `hyper`, and, if it does, then I don't think that would be desirable.

**Review by Geal — APPROVED** [review]

**Review by garypen — APPROVED** [review]

**Review by BrynCooke — APPROVED** [review]

**Pull request merged**